pkg/http/proxy.go

```diff
@@ -31,15 +32,38 @@ func NewHeaderPruningReverseProxy(targetHost string, headersToRemove []string) *
 			req.URL.Scheme = "http"
 			req.URL.Host = targetHost
 
-			// Copied from httputil.NewSingleHostReverseProxy.
-			if _, ok := req.Header[network.UserAgentKey]; !ok {
-				// explicitly disable User-Agent so it's not set to default value
-				req.Header.Set(network.UserAgentKey, "")
-			}
+			handleHeaders(req, headersToRemove)
+		},
+	}
+}
 
-			for _, h := range headersToRemove {
-				req.Header.Del(h)
-			}
+// NewPassthroughLbHeaderPruningReverseProxy returns a httputil.ReverseProxy that proxies
+// requests to the given targetHost after removing the headersToRemove.
+// It replaces the Host header with the given spoofedHost and sets the Knative
```

Comment:
maybe `hostOverride` instead of spoofed host? Im not sure we're really spoofing so much as explicitly setting here